cylc/flow/scripts/lint.py

```diff
@@ -1340,6 +1343,13 @@ def get_option_parser() -> COP:
         action='store_true',
         default=False,
     )
+    parser.add_option(
+        '--rule', '-R',
+        help='Carry out only specific checks.',
+        choices=list(parse_checks(['style', '728']).keys()),
```

Comment:
So `--rule` can be one of `style` or `728`.

AND `--ruleset` can be one of `style`, `728` or `all`.

What's the point of `--rule`?

Reply:
The `--select` / `--ignore` interface of `ruff` is a good gauge of how to go about implementing rule selection.

```
--select=R  # select all Rxxx rules
--select=R001  # select only R001
--select=R --ignore=R001  # select all R other than R001
--select=R001,R002  # select both R001 and R002
--select=ALL  # magic
```

Reply:
> AND --ruleset can be one of style, 728 or all.

No - rulest can be any of the items in the parsed checks for both rulesets currently available because 

```python
>>> parse_checks(['style', '728'].keys()
DictKeys('S001', 'S002' ... ... 'R999')

```